internal/pkg/cli/pipeline_deploy.go

```diff
@@ -258,13 +270,70 @@ func (o *deployPipelineOpts) Execute() error {
 		PermissionsBoundary: o.app.PermissionsBoundary,
 	}
 
+	if o.showDiff {
+		tpl, err := o.pipelineStackConfig(deployPipelineInput).Template()
+		if err != nil {
+			return fmt.Errorf("generate the new template for deploy diff: %w", err)
+		}
+		if err = o.diff(deployPipelineInput, tpl); err != nil {
+			var errHasDiff *errHasDiff
+			if !errors.As(err, &errHasDiff) {
+				return err
+			}
+		}
+		if !o.skipConfirmation {
+			contd, err := o.prompt.Confirm(continueDeploymentPrompt, "")
+			if err != nil {
+				return fmt.Errorf("ask whether to continue with the deployment: %w", err)
+			}
+			if !contd {
+				return nil
+			}
+		}
+	}
 	if err := o.deployPipeline(deployPipelineInput); err != nil {
 		return err
 	}
-
 	return nil
 }
 
+// PipelineDeployDiff returns the stringified diff of the template against the deployed template of the pipeline.
+func (o *deployPipelineOpts) pipelineDeployDiff(in *deploy.CreatePipelineInput, template string) (string, error) {
```

Comment:
Can we do something like env to move this to the `cli/deploy` pkg?

Reply:
This would need a big refactor, Probably we will have a separate refactor for this.